authors/apps/articles/views.py

```diff
@@ -89,3 +90,123 @@ def delete(self, request, slug):
         return Response({
             'errors': 'sorry article with that slug doesnot exist'
         }, status=status.HTTP_404_NOT_FOUND)
+
+
+class LikeArticleAPIView(APIView):
+    """
+    Expose endpoints allowing a user to like an article and undo a like
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, slug):
+        """
+        Add a user to the list of people that liked an article
+        :param request: The incoming request object
+        :param slug: The expected article slug
+        :return: A response indicating success for the action, or the appropriate error on failure
+        """
+        article = model_helpers.get_single_article_using_slug(slug)
+
+        if article:
+            article.disliked_by.remove(request.user)
+            article.liked_by.add(request.user)
+            return Response({
+                'message': 'liked'
+            })
+
+        return Response({
+            'errors': 'article with that slug does not exist'
+        }, status=status.HTTP_404_NOT_FOUND)
+
+    def delete(self, request, slug):
+        """
+        Remove a user from the list of people that liked an article
+        """
+        article = model_helpers.get_single_article_using_slug(slug)
+
+        if article:
+            article.liked_by.remove(request.user)
+            return Response({
+                'message': 'like undone'
+            })
+
+        return Response({
+            'errors': 'article with that slug does not exist'
+        }, status=status.HTTP_404_NOT_FOUND)
+
+
+class DislikeArticleAPIView(APIView):
```

Comment:
@Karuhanga I think as a team we agreed to use generic views.

Reply:
Thanks for the heads up.
Changed this.